File: JANUARY/15-01-2026/2943.cpp
class Solution {
public:
    int maximizeSquareHoleArea(int n, int m, vector<int>& hBars, vector<int>& vBars) {
        auto longest = [](vector<int>& a) {
            sort(a.begin(), a.end());
            int best = 1, cur = 1;
            for (int i = 1; i < a.size(); i++) {
                if (a[i] == a[i - 1] + 1) cur++;
                else cur = 1;
                best = max(best, cur);
            }
            return best + 1;
        };
        int h = longest(hBars);
        int v = longest(vBars);
        int side = min(h, v);
        return side * side;
    }
};

File: JANUARY/15-01-2026/2943.rs
impl Solution {
    pub fn maximize_square_hole_area(n: i32, m: i32, mut h_bars: Vec<i32>, mut v_bars: Vec<i32>) -> i32 {
        fn longest(mut a: Vec<i32>) -> i32 {
            a.sort();
            let mut best = 1;
            let mut cur = 1;
            for i in 1..a.len() {
                if a[i] == a[i - 1] + 1 {
                    cur += 1;
                } else {
                    cur = 1;
                }
                if cur > best {
                    best = cur;
                }
            }
            best + 1
        }

        let side = std::cmp::min(longest(h_bars), longest(v_bars));
        side * side
    }
}